apps/studio/src/features/editing-experience/components/RootStateDrawer.tsx

```diff
@@ -223,6 +237,60 @@ export default function RootStateDrawer() {
     pageLayout !== "index" &&
     pageLayout !== "collection"
 
+  const invalidBlockIndexes = new Set<number>()
+  validateFn(savedPageState)
+  for (const error of validateFn.errors ?? []) {
+    if (error.instancePath.startsWith("/content")) {
+      const pathAfterContent = error.instancePath.split("/content/")[1]
+      if (pathAfterContent) {
+        const pathParts = pathAfterContent.split("/")
+        const blockIndex = pathParts[0]
+        if (blockIndex && !isNaN(parseInt(blockIndex))) {
+          invalidBlockIndexes.add(parseInt(blockIndex))
+        }
+      }
+    }
+  }
+
+  const FixedBlock = () => {
+    // Assuming only one fixedBlock can exist at a time for now
+    const fixedBlock = savedPageState.content[0]
+
+    if (isHeroFixedBlock) {
+      const subSchema = getComponentSchema({ component: "hero" })
+      const validateFn = ajv.compile<IsomerComponent>(subSchema)
+      const isValid = validateFn(fixedBlock)
+      return (
+        <BaseBlock
+          onClick={() => {
+            setCurrActiveIdx(0)
+            setDrawerState({ state: "heroEditor" })
+          }}
+          label="Hero banner"
+          description="Title, subtitle, and Call-to-Action"
+          icon={TYPE_TO_ICON.hero}
+          isInvalid={!isValid}
+        />
+      )
+    }
```

Comment:
mostly just extracting and moving the rendering logic here, so that i can add the hero-component validation